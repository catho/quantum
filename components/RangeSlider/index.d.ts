import React from 'react';

export interface RangeSliderProps {
    step?: number;
    max?: number;
    min?: number;
    disabled?: boolean;
    onChange?: React.ChangeEventHandler<HTMLInputElement>;
    onChangeCommitted?: React.ChangeEventHandler<HTMLInputElement>;
    marks?: Array<{
        value?: string | number;
        label?: string;
    }>;
    valueLabelDisplay?: 'auto' | 'on' | 'off';
    track?: 'normal' | false | 'inverted';
    'aria-labelledby'?: string;
    tipFormatter?: (value?: number, index?: number) => string;
    value?: number | { from: number; to: number };
    defaultValue?: number | { from: number; to: number };
    theme?: {
        spacing?: object;
        colors?: object;
        baseFontSize?: number;
    };
}

export default class RangeSlider extends React.Component<RangeSliderProps> {}
