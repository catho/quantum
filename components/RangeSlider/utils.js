const mountMarksByArray = (values, labelFunction) =>
  values.map(value => ({
    value,
    label: labelFunction ? labelFunction(value) : value,
  }));

export default {
  mountMarksByArray,
};
