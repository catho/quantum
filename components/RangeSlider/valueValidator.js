const valueValidator = (props, propName, componentName) => {
  const { [propName]: value } = props;
  const { max, min } = props;
  const errorMessage = `${propName} prop supplied to ${componentName} is out of bounds. The ${propName} prop must be between ${min} and ${max}.`;
  const isNumber = val => typeof val === 'number';

  if (typeof value === 'object') {
    const { from, to } = value;

    if (!isNumber(from) || !isNumber(to)) {
      return new Error(
        `The 'from' and 'to' keys from ${propName} prop must be a Number.`,
      );
    }

    if (from < min || to > max) {
      return new Error(errorMessage);
    }

    return null;
  }

  if (!isNumber(value) && value !== undefined) {
    return new Error(`${propName} prop must be a Number.`);
  }

  if (value < min || value > max) {
    return new Error(errorMessage);
  }

  return null;
};

export default valueValidator;
