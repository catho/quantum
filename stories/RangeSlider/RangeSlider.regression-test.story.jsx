import React from 'react';
import { storiesOf } from '@storybook/react';
import { Row, Col } from '../../components';
import rangeSliderUtils from '../../components/RangeSlider/utils';

import {
  Slider,
  TipFormatter,
  Range,
  MinMax,
  BasicDisabledSlider,
} from './examples';

const sliderProps = {
  valueLabelDisplay: 'on',
};

const { mountMarksByArray } = rangeSliderUtils;

const components = [
  { name: 'default', el: <Slider {...sliderProps} /> },
  {
    name: 'with marks',
    el: <Slider marks={mountMarksByArray([0, 100])} {...sliderProps} />,
  },
  { name: 'TipFormatter', el: <TipFormatter {...sliderProps} /> },
  {
    name: 'TipFormatter with marks',
    el: (
      <TipFormatter
        {...sliderProps}
        marks={mountMarksByArray([0, 50, 100], v => `R$ ${v}`)}
      />
    ),
  },
  { name: 'Range', el: <Range {...sliderProps} /> },
  { name: 'MinMax', el: <MinMax {...sliderProps} /> },
  { name: 'BasicDisabledSlider', el: <BasicDisabledSlider {...sliderProps} /> },
];

const renderComponents = component => (
  <Row style={{ padding: '50px 180px' }}>
    <Col>{component.el}</Col>
  </Row>
);

components.forEach(component => {
  storiesOf('RangeSlider', module).add(`${component.name}`, () =>
    renderComponents(component),
  );
});
