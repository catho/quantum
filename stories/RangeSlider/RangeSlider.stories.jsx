import { useState } from 'react';
import { RangeSlider } from '../../components';

export default {
  title: 'Forms/RangeSlider',
  component: RangeSlider,
};

const marks = [
  {
    value: 0,
    label: '0 km',
  },
  {
    value: 50,
    label: '50 km',
  },
  {
    value: 100,
    label: '100 km',
  },
];

const Template = (args) => <RangeSlider {...args} />;

export const Default = Template.bind({});

export const WithMarks = Template.bind({});
WithMarks.args = {
  marks,
};

export const Disabled = Template.bind({});
Disabled.args = {
  disabled: true,
};

export const MinAndMax = Template.bind({});
MinAndMax.args = {
  min: 30,
  max: 50,
  defaultValue: 40,
};

export const RangeUsage = Template.bind({});
RangeUsage.args = {
  defaultValue: {
    from: 20,
    to: 80,
  },
};

export const CustomTooltipText = Template.bind({});
CustomTooltipText.args = {
  tipFormatter: (value) => `R$ ${value}`,
};

export const ControlledExample = () => {
  const [value, setValue] = useState(0);

  const handleChange = (event, updatedValue) => {
    setValue(updatedValue);
  };

  return (
    <>
      Slider value: {value}
      <RangeSlider defaultValue={value} onChangeCommitted={handleChange} />
    </>
  );
};
