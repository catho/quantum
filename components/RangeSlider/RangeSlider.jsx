import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import MaterialSlider from '@material-ui/core/Slider';
import withStyles from '@material-ui/core/styles/withStyles';

import {
  colors,
  spacing,
  baseFontSize as defaultBaseFontSize,
} from '../shared/theme';
import { hexToRgba } from '../shared';
import valueValidator from './valueValidator';

const SliderWrapper = styled.div`
  width: auto;
  ${({
    theme: {
      spacing: { large },
    },
  }) => `
      padding: 0 ${large / 2}px;
    `}
`;

const RangeSlider = props => {
  const {
    'aria-labelledby': ariaLabelledby,
    defaultValue,
    disabled,
    marks,
    max,
    min,
    onChange,
    onChangeCommitted,
    step,
    theme,
    tipFormatter,
    track,
    value,
    valueLabelDisplay,
  } = props;

  const {
    colors: { neutral, primary },
    spacing: { xsmall, small, large },
    baseFontSize,
  } = theme;

  const arrowSize = xsmall - 1;

  const halfMeasure = '50%';

  const StyledSlider = withStyles({
    root: {
      color: primary[600],
      height: 8,
    },
    thumb: {
      backgroundColor: disabled ? neutral[500] : primary[700],
      border: 'none',
      height: `${large}px !important`,
      width: `${large}px !important`,
      marginTop: `-${xsmall}px !important`,
      marginLeft: `-${large / 2}px !important`,
      '&:focus,&:hover': {
        '& > *': {
          transform: `scale(1) translateX(-${halfMeasure}) !important`,
        },
      },
    },
    valueLabel: {
      width: 'auto',
      top: '-50px',
      left: halfMeasure,
      transform:
        valueLabelDisplay === 'on'
          ? `scale(1) translateX(-${halfMeasure}) !important`
          : 'scale(0)',
      '&:focus,&:hover': {
        transform: `scale(1) translateX(-${halfMeasure}) !important`,
      },
      '&::after': {
        borderLeft: `${arrowSize}px solid transparent`,
        borderRight: `${arrowSize}px solid transparent`,
        borderTop: `${arrowSize}px solid ${neutral[700]}`,
        bottom: `-${arrowSize}px`,
        content: `" "`,
        height: 0,
        left: halfMeasure,
        position: 'absolute',
        transform: `translateX(-${halfMeasure}) !important`,
        width: 0,
      },
      '& > *': {
        background: neutral[700],
        color: neutral[0],
        fontSize: `${baseFontSize}px`,
        padding: `${xsmall}px ${small}px`,
        borderRadius: '4px',
        height: `${large}px`,
        transform: 'none',
        width: 'auto',
        whiteSpace: 'nowrap',

        '& > *': {
          width: 'auto',
          transform: 'none',
        },
      },
    },
    mark: {
      display: 'none',
    },
    markLabel: {
      marginTop: `${xsmall}px`,
    },
    track: {
      height: 8,
      borderRadius: 4,
      backgroundColor: hexToRgba(primary[700], 0.5),
    },
    rail: {
      height: 8,
      borderRadius: 4,
    },
  })(MaterialSlider);

  StyledSlider.displayName = 'SliderComponent';

  const formatedValue =
    typeof value === 'object' ? [value.from, value.to] : value;
  const formatedDefaultValue =
    typeof defaultValue === 'object'
      ? [defaultValue.from, defaultValue.to]
      : defaultValue;

  return (
    <SliderWrapper theme={theme}>
      <StyledSlider
        aria-labelledby={ariaLabelledby}
        defaultValue={disabled ? 0 : formatedDefaultValue}
        disabled={disabled}
        getAriaLabel={tipFormatter}
        getAriaValueText={tipFormatter}
        marks={marks}
        max={max}
        min={min}
        onChange={onChange}
        onChangeCommitted={onChangeCommitted}
        ref={React.createRef()}
        step={step}
        track={track}
        value={disabled ? 0 : formatedValue}
        valueLabelDisplay={disabled ? 'off' : valueLabelDisplay}
        valueLabelFormat={tipFormatter}
      />
    </SliderWrapper>
  );
};

SliderWrapper.displayName = 'RangeSliderWrapper';

RangeSlider.defaultProps = {
  step: 1,
  max: 100,
  min: 0,
  value: undefined,
  defaultValue: 30,
  marks: undefined,
  disabled: false,
  onChange: () => {},
  onChangeCommitted: () => {},
  tipFormatter: value => value.toString(),
  track: 'normal',
  'aria-labelledby': undefined,
  valueLabelDisplay: 'auto',
  theme: {
    spacing,
    colors,
    baseFontSize: defaultBaseFontSize,
  },
};

RangeSlider.propTypes = {
  /** The number that the slider will use as interval of each value */
  step: PropTypes.number,
  /** The max value possible in slider */
  max: PropTypes.number,
  /** The min value possible in slider */
  min: PropTypes.number,
  disabled: PropTypes.bool,
  /** Trigger a function on value change, evict to use controlled values, because the component will re-render before change the value */
  onChange: PropTypes.func,
  /** Trigger a function after value completely changes */
  onChangeCommitted: PropTypes.func,
  /** The min value possible in slider */
  marks: PropTypes.arrayOf(
    PropTypes.shape({
      value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
      label: PropTypes.string,
    }),
  ),
  /* *Set if the tooltip label is by default visible. 'auto' the value label will display when the thumb is hovered or focused. 'on' will display persistently. 'off' will never display. */
  valueLabelDisplay: PropTypes.oneOf(['auto', 'on', 'off']),
  /** The track presentation:- normal the track will render a bar representing the slider value. - inverted the track will render a bar representing the remaining slider value. - false the track will render without a bar. */
  track: PropTypes.oneOf(['normal', false, 'inverted']),
  'aria-labelledby': PropTypes.string,
  /** RangeSlider will pass its value to tipFormatter, display its value in Tooltip */
  tipFormatter: PropTypes.func,
  /** It receives a Number to display a slider or an Object with from and to properties to display a range. Example: `value={10}` or `value={{ from: 20, to: 40 }}` */
  value: valueValidator,
  defaultValue: valueValidator,
  theme: PropTypes.shape({
    spacing: PropTypes.object,
    colors: PropTypes.object,
    baseFontSize: PropTypes.number,
  }),
};

export default RangeSlider;
