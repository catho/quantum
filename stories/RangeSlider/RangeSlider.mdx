import { Meta, Canvas, Story, ArgsTable } from '@storybook/addon-docs';

import { RangeSlider } from '../../components';

import { Header } from '../shared';

import { 
  Default,
  WithMarks,
  Disabled,
  MinAndMax,
  RangeUsage,
  CustomTooltipText,
  ControlledExample,
} from './RangeSlider.stories.jsx';

<Meta title="Forms/RangeSlider" component={RangeSlider}  />

<Header title="RangeSlider">
  RangeSlider is a component to select a value from a predefined range.
  Through component's props, it's possible to set the minimum and maximum
  values, format tooltip, label texts, or disable the
  component itself.
</Header>

- [Importing and usage](#importing)
- [API](#api)
- [Examples](#examples)

## <a id="importing"></a>Importing and usage

```js
import RangeSlider from '@catho/quantum/RangeSlider';
```

<Canvas>
  <Story of={Default} />
</Canvas>

## <a id="api"></a>API

<ArgsTable of={RangeSlider} />

## <a id="examples"></a>Examples

### With Marks
<Canvas>
  <Story of={WithMarks} />
</Canvas>

### Disabled
<Canvas>
  <Story of={Disabled} />
</Canvas>

### Min & Max

You can limit the available range to be selected.

<Canvas>
  <Story of={MinAndMax} />
</Canvas>

### Range usage

Set value prop with an object with "from" and "to" properties.

<Canvas>
  <Story of={RangeUsage} />
</Canvas>

### Custom tooltip texts

You can format tooltip text through `tipFormatter` prop.
It receives a function and should return a string. *You can hover example to see.*

<Canvas>
  <Story of={CustomTooltipText} />
</Canvas>

### Controlled example

It needs a [controlled component](https://reactjs.org/docs/react-component.html#setstate) to change the RangeSlider value.
The same is applied when an object is used in value prop.

<Canvas>
  <Story of={ControlledExample} />
</Canvas>