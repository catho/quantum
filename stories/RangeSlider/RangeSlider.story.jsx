import React from 'react';
import styled from 'styled-components';
import { storiesOf } from '@storybook/react';
import {
  Title,
  Heading,
  TabbedView,
  Tab,
  SimpleHighlight,
  CodeExample,
  AutoPropsApi,
} from '@catho/quantum-storybook-ui';
import { StoryContainer } from '@catho/quantum-storybook-ui/dist/components';
import { Col, Row } from '../../components/Grid';
import RangeSlider from '../../components/RangeSlider';
import List from '../../components/List';
import * as Samples from './examples';

const importSlider = `
  import { RangeSlider } from '@catho/quantum';

  const marks = [
    {
      value: 10,
      label: '10 km',
    },
    {
      value: 50,
      label: '50 km',
    },
    {
      value: 100,
      label: '100 km',
    }
  ]
`;

const marks = [
  {
    value: 0,
    label: '0 km',
  },
  {
    value: 50,
    label: '50 km',
  },
  {
    value: 100,
    label: '100 km',
  },
];

const ColSample = styled(Col)`
  padding-top: 15px;
`;

storiesOf('Forms', module).add('RangeSlider', () => (
  <>
    <Heading name="RangeSlider">
      RangeSlider is a component to select a value from a predefined range.
      Through component props it&apos;s possible to set the minimum and maximum
      values to be selected, format tooltip and label texts or disable the
      component itself.
    </Heading>

    <TabbedView>
      <Tab title="Usage">
        <StoryContainer>
          <Title as="h2">Importing RangeSlider</Title>
          <SimpleHighlight>{importSlider}</SimpleHighlight>

          <Row>
            <Col xsmall={4} small={8} medium={12}>
              <p>
                You can use some helper props to display the correct information
                desired.
              </p>
              <List>
                <List.Item>
                  <strong>min</strong>: accepted min value (default: 0)
                </List.Item>
                <List.Item>
                  <strong>max</strong>: accepted max value (default: 100)
                </List.Item>
                <List.Item>
                  <strong>tipFormatter</strong>: format tooltip text
                </List.Item>
              </List>
            </Col>
          </Row>
          <Title as="h2">RangeSlider usage</Title>
          <p>Just set value prop with a Number value.</p>
          <Row>
            <Col xsmall={2} small={4} medium={6}>
              <CodeExample
                code={Samples.BasicRangeSlider.code}
                showTitle={false}
              />
              <p>With marks</p>
              <CodeExample
                code={Samples.BasicRangeSlider.codeWithMarks}
                showTitle={false}
              />
            </Col>
            <ColSample xsmall={2} small={4} medium={6}>
              <Samples.BasicRangeSlider />
              <p>With marks</p>
              <Samples.BasicRangeSlider marks={marks} />
            </ColSample>
          </Row>
          <Title as="h2">Disable</Title>
          <Row>
            <Col xsmall={2} small={4} medium={6}>
              <CodeExample
                code={Samples.BasicDisabledRangeSlider.code}
                showTitle={false}
              />
            </Col>
            <ColSample xsmall={2} small={4} medium={6}>
              <Samples.BasicDisabledRangeSlider />
            </ColSample>
          </Row>
          <Title as="h2">Min & Max props</Title>
          <p>You can limit the available range to be selected</p>
          <Row>
            <Col xsmall={2} small={4} medium={6}>
              <CodeExample code={Samples.MinMax.code} showTitle={false} />
            </Col>
            <ColSample xsmall={2} small={4} medium={6}>
              <Samples.MinMax />
            </ColSample>
          </Row>
          <Title as="h2">Range usage</Title>
          <p>
            Set value prop with an Object with &quot;to&quot; and
            &quot;from&quot; properties.
          </p>
          <Row>
            <Col xsmall={2} small={4} medium={6}>
              <CodeExample code={Samples.Range.code} showTitle={false} />
            </Col>
            <ColSample xsmall={2} small={4} medium={6}>
              <Samples.Range />
            </ColSample>
          </Row>
          <Title as="h2">Formatting tooltip text</Title>
          <p>
            You can format tooltip text through <strong>tipFormatter</strong>{' '}
            prop.
          </p>
          <p>
            It receives a function and should return a string.{' '}
            <em>
              <small>(hover {'<RangeSlider />'} to see.)</small>
            </em>
          </p>
          <Row>
            <Col xsmall={2} small={4} medium={6}>
              <CodeExample code={Samples.TipFormatter.code} showTitle={false} />
            </Col>
            <ColSample xsmall={2} small={4} medium={6}>
              <Samples.TipFormatter />
            </ColSample>
          </Row>

          <Title as="h2">Controlled RangeSlider</Title>
          <p>
            it&apos;s needed an
            <a
              href="https://reactjs.org/docs/forms.html#controlled-components"
              rel="noopener noreferrer"
              target="_blank"
              title="controlled component"
            >
              controlled component
            </a>
            to change the {'<RangeSlider />'} value.
          </p>
          <p>The same is applied when is passed an object to value prop.</p>
          <Row>
            <Col xsmall={2} small={4} medium={6}>
              <CodeExample
                code={Samples.ControlledRangeSlider.code}
                showTitle={false}
              />
            </Col>
            <ColSample xsmall={2} small={4} medium={6}>
              <Samples.ControlledRangeSlider />
            </ColSample>
          </Row>
        </StoryContainer>
      </Tab>
      <Tab title="API">
        <AutoPropsApi component={RangeSlider} />
      </Tab>
    </TabbedView>
  </>
));
