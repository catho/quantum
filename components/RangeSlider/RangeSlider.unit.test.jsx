import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import RangeSlider from './RangeSlider';

describe('<RangeSlider />', () => {
  describe('Snapshots', () => {
    it('should match the snapshot when have just one handle', () => {
      expect(toJson(shallow(<RangeSlider value={10} />))).toMatchSnapshot();
    });

    it('should match the snapshot when have two handles', () => {
      expect(
        toJson(shallow(<RangeSlider value={{ from: 10, to: 30 }} />)),
      ).toMatchSnapshot();
    });

    it('should match the snapshot when <RangeSlider /> is disabled', () => {
      expect(toJson(shallow(<RangeSlider disabled />))).toMatchSnapshot();
    });
  });

  describe('tipFormatter prop', () => {
    it('should format Tooltip text and display it correctly when have just one handle', () => {
      const range = mount(
        <RangeSlider value={10} tipFormatter={value => `R$ ${value}`} />,
      );
      const tooltip = range.find('ValueLabel');

      expect(tooltip.text()).toBe('R$ 10');
    });

    it('should format Tooltip text and display it correctly when have two handles', () => {
      const range = mount(
        <RangeSlider
          value={{ from: 10, to: 40 }}
          tipFormatter={value => `${value}km`}
        />,
      );
      const tooltip = range.find('ValueLabel');

      expect(tooltip.at(0).text()).toBe('10km');
      expect(tooltip.at(1).text()).toBe('40km');
    });
  });

  describe('onChange prop', () => {
    it('should call it with currently slider value', () => {
      const onChangeMock = jest.fn();
      const range = shallow(<RangeSlider value={10} onChange={onChangeMock} />);
      const slider = range.find('SliderComponent');

      slider.simulate('change', 20);

      expect(onChangeMock).toBeCalledWith(20);
    });

    it('should call it with currently range value', () => {
      const onChangeMock = jest.fn();
      const range = shallow(
        <RangeSlider value={{ from: 0, to: 100 }} onChange={onChangeMock} />,
      );
      const slider = range.find('SliderComponent');

      slider.simulate('change', { from: 20, to: 80 });

      expect(onChangeMock.mock.calls.length).toBe(1);
    });
  });

  describe('onChangeCommited prop', () => {
    it('should call it with currently slider value', () => {
      const onChangeCommitedMock = jest.fn();
      const range = shallow(
        <RangeSlider value={10} onChangeCommited={onChangeCommitedMock} />,
      );
      const slider = range.find('SliderComponent');

      slider.simulate('change', 20);

      setTimeout(() => {
        expect(onChangeCommitedMock).toBeCalledWith(20);
      }, 1000);
    });

    it('should call it with currently range value', () => {
      const onChangeCommitedMock = jest.fn();
      const range = shallow(
        <RangeSlider
          value={{ from: 0, to: 100 }}
          onChangeCommited={onChangeCommitedMock}
        />,
      );
      const slider = range.find('SliderComponent');

      slider.simulate('change', { from: 20, to: 80 });

      setTimeout(() => {
        expect(onChangeCommitedMock.mock.calls.length).toBe(1);
      }, 1000);
    });
  });

  describe('Value prop validator', () => {
    global.console = {
      error: jest.fn(),
    };

    beforeEach(global.console.error.mockReset);

    describe('with a single value', () => {
      it('should not call console.error when it is valid', () => {
        mount(<RangeSlider value={0} />);
        expect(global.console.error).not.toHaveBeenCalled();
      });

      it('should call console.error when it is not a number', () => {
        mount(<RangeSlider value="1" />);
        expect(global.console.error).toHaveBeenCalled();
      });

      it('should call console.error when it is greater then max prop', () => {
        mount(<RangeSlider max={40} value={50} />);
        expect(global.console.error).toHaveBeenCalled();
      });

      it('should call console.error when it is less then min prop', () => {
        mount(<RangeSlider min={40} value={30} />);
        expect(global.console.error).toHaveBeenCalled();
      });
    });

    describe('with a range value', () => {
      it('should not call console.error when both is valid', () => {
        mount(<RangeSlider value={{ from: 0, to: 100 }} />);
        expect(global.console.error).not.toHaveBeenCalled();
      });

      it('should call console.error when one of them are not a number', () => {
        mount(<RangeSlider value={{ from: '0', to: 100 }} />);
        expect(global.console.error).toHaveBeenCalled();

        mount(<RangeSlider value={{ from: 0, to: '100' }} />);
        expect(global.console.error).toHaveBeenCalled();
      });

      it('should call console.error when "to" value is greater then max prop', () => {
        mount(<RangeSlider value={{ from: 0, to: 100 }} max={90} />);
        expect(global.console.error).toHaveBeenCalled();
      });

      it('should call console.error when "from" value is less then min prop', () => {
        mount(<RangeSlider value={{ from: 0, to: 100 }} min={10} />);
        expect(global.console.error).toHaveBeenCalled();
      });
    });
  });
});
