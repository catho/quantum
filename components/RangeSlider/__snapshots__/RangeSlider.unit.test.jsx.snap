// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RangeSlider /> Snapshots should match the snapshot when <RangeSlider /> is disabled 1`] = `
<RangeSliderWrapper
  theme={
    Object {
      "baseFontSize": 16,
      "colors": Object {
        "error": Object {
          "100": "#fff5f5",
          "300": "#f88e86",
          "500": "#ec584e",
          "700": "#e91b0c",
          "900": "#ac001a",
        },
        "neutral": Object {
          "0": "#ffffff",
          "100": "#f2f2f2",
          "1000": "#000000",
          "300": "#e0e0e0",
          "500": "#999999",
          "700": "#424242",
          "900": "#191919",
        },
        "primary": Object {
          "100": "#E5EDFC",
          "300": "#89AAE7",
          "500": "#0CC0EA",
          "700": "#1250C4",
          "900": "#002F7B",
        },
        "secondary": Object {
          "100": "#F7B0C8",
          "300": "#F278A1",
          "500": "#E91E63",
          "700": "#DE0059",
          "900": "#9E003F",
        },
        "success": Object {
          "100": "#dcedc8",
          "300": "#c9eda2",
          "500": "#7ed321",
          "700": "#3b610f",
          "900": "#004D40",
        },
        "warning": Object {
          "100": "#ffefd6",
          "300": "#ffda6a",
          "500": "#ffc107",
          "700": "#f09100",
          "900": "#d14900",
        },
      },
      "spacing": Object {
        "large": 24,
        "medium": 16,
        "small": 12,
        "xlarge": 32,
        "xsmall": 8,
        "xxlarge": 40,
        "xxsmall": 4,
        "xxxlarge": 48,
        "xxxsmall": 2,
      },
    }
  }
>
  <SliderComponent
    defaultValue={0}
    disabled={true}
    getAriaLabel={[Function]}
    getAriaValueText={[Function]}
    max={100}
    min={0}
    onChange={[Function]}
    onChangeCommitted={[Function]}
    step={1}
    track="normal"
    value={0}
    valueLabelDisplay="off"
    valueLabelFormat={[Function]}
  />
</RangeSliderWrapper>
`;

exports[`<RangeSlider /> Snapshots should match the snapshot when have just one handle 1`] = `
<RangeSliderWrapper
  theme={
    Object {
      "baseFontSize": 16,
      "colors": Object {
        "error": Object {
          "100": "#fff5f5",
          "300": "#f88e86",
          "500": "#ec584e",
          "700": "#e91b0c",
          "900": "#ac001a",
        },
        "neutral": Object {
          "0": "#ffffff",
          "100": "#f2f2f2",
          "1000": "#000000",
          "300": "#e0e0e0",
          "500": "#999999",
          "700": "#424242",
          "900": "#191919",
        },
        "primary": Object {
          "100": "#E5EDFC",
          "300": "#89AAE7",
          "500": "#0CC0EA",
          "700": "#1250C4",
          "900": "#002F7B",
        },
        "secondary": Object {
          "100": "#F7B0C8",
          "300": "#F278A1",
          "500": "#E91E63",
          "700": "#DE0059",
          "900": "#9E003F",
        },
        "success": Object {
          "100": "#dcedc8",
          "300": "#c9eda2",
          "500": "#7ed321",
          "700": "#3b610f",
          "900": "#004D40",
        },
        "warning": Object {
          "100": "#ffefd6",
          "300": "#ffda6a",
          "500": "#ffc107",
          "700": "#f09100",
          "900": "#d14900",
        },
      },
      "spacing": Object {
        "large": 24,
        "medium": 16,
        "small": 12,
        "xlarge": 32,
        "xsmall": 8,
        "xxlarge": 40,
        "xxsmall": 4,
        "xxxlarge": 48,
        "xxxsmall": 2,
      },
    }
  }
>
  <SliderComponent
    defaultValue={30}
    disabled={false}
    getAriaLabel={[Function]}
    getAriaValueText={[Function]}
    max={100}
    min={0}
    onChange={[Function]}
    onChangeCommitted={[Function]}
    step={1}
    track="normal"
    value={10}
    valueLabelDisplay="auto"
    valueLabelFormat={[Function]}
  />
</RangeSliderWrapper>
`;

exports[`<RangeSlider /> Snapshots should match the snapshot when have two handles 1`] = `
<RangeSliderWrapper
  theme={
    Object {
      "baseFontSize": 16,
      "colors": Object {
        "error": Object {
          "100": "#fff5f5",
          "300": "#f88e86",
          "500": "#ec584e",
          "700": "#e91b0c",
          "900": "#ac001a",
        },
        "neutral": Object {
          "0": "#ffffff",
          "100": "#f2f2f2",
          "1000": "#000000",
          "300": "#e0e0e0",
          "500": "#999999",
          "700": "#424242",
          "900": "#191919",
        },
        "primary": Object {
          "100": "#E5EDFC",
          "300": "#89AAE7",
          "500": "#0CC0EA",
          "700": "#1250C4",
          "900": "#002F7B",
        },
        "secondary": Object {
          "100": "#F7B0C8",
          "300": "#F278A1",
          "500": "#E91E63",
          "700": "#DE0059",
          "900": "#9E003F",
        },
        "success": Object {
          "100": "#dcedc8",
          "300": "#c9eda2",
          "500": "#7ed321",
          "700": "#3b610f",
          "900": "#004D40",
        },
        "warning": Object {
          "100": "#ffefd6",
          "300": "#ffda6a",
          "500": "#ffc107",
          "700": "#f09100",
          "900": "#d14900",
        },
      },
      "spacing": Object {
        "large": 24,
        "medium": 16,
        "small": 12,
        "xlarge": 32,
        "xsmall": 8,
        "xxlarge": 40,
        "xxsmall": 4,
        "xxxlarge": 48,
        "xxxsmall": 2,
      },
    }
  }
>
  <SliderComponent
    defaultValue={30}
    disabled={false}
    getAriaLabel={[Function]}
    getAriaValueText={[Function]}
    max={100}
    min={0}
    onChange={[Function]}
    onChangeCommitted={[Function]}
    step={1}
    track="normal"
    value={
      Array [
        10,
        30,
      ]
    }
    valueLabelDisplay="auto"
    valueLabelFormat={[Function]}
  />
</RangeSliderWrapper>
`;
